lib/activeadmin/searchable_select/select_input_extension.rb

```diff
@@ -57,23 +57,23 @@ def all_options_collection
       end
 
       def selected_value_collection
-        [selected_value_option].compact
+        selected_value_options ? selected_value_options.compact : []
       end
 
-      def selected_value_option
-        option_for_record(selected_record) if selected_record
+      def selected_value_options
+        selected_records.collect{|s| option_for_record(s)} if selected_records
       end
 
       def option_for_record(record)
         [option_collection.display_text(record), record.id]
       end
 
-      def selected_record
-        @selected_record ||=
-          selected_value && option_collection_scope.find_by_id(selected_value)
+      def selected_records
+        @selected_records ||=
+          selected_values && option_collection_scope.where(id: selected_values)
```

Comment:
If we change this to return an empty array instead of `nil`, the `if` in `selected_value_options` should no longer be needed. `selected_value_options` can then be merged with `selected_value_collection` since the `compact` was only needed in the old implementation to handle the case when `selected_value_option` returned nil.